packages/x-components/src/components/modals/base-modal.vue

```diff
@@ -1,14 +1,15 @@
 <template>
-  <component :is="animation">
+  <component :is="animation" @beforeEnter="showOverlay = false" @afterEnter="showOverlay = true">
```

Comment:
Like you said, this needs a rethink in the future. We want to animate the content div, not the `x-modal` one, probably leaving this wrapper div is a trade of that I would happily do

Reply:
That tradeoff breaks my mobile layout.

We will come back to this in other moment